numba/np/ufunc/dufunc.py

```diff
@@ -778,6 +779,136 @@ def impl_axis_none(ufunc,
                 # ufunc(array, axis=0)
                 return impl_nd_axis_int
 
+    def _install_ufunc_reduceat(self, template) -> None:
+        at = types.Function(template)
+
+        @overload_method(at, 'reduceat')
+        def ol_reduceat(ufunc, array, indices, axis=0, dtype=None, out=None):
+
+            warnings.warn("ufunc.reduceat feature is experimental",
+                          category=errors.NumbaExperimentalFeatureWarning)
+
+            if self.ufunc.nin != 2:
+                msg = 'reduceat only supported for binary functions'
+                raise errors.NumbaTypeError(msg)
+
+            if not numpy_support.type_can_asarray(array):
+                msg = 'The first argument "array" must be array-like'
+                raise errors.NumbaTypeError(msg)
+
+            if not numpy_support.type_can_asarray(indices):
+                msg = 'The second argument "indices" must be array-like'
+                raise errors.NumbaTypeError(msg)
+
+            if not isinstance(axis, (types.Integer, types.Omitted,
+                                     types.IntegerLiteral, int)):
+                msg = '"axis" must be an integer'
+                raise errors.NumbaTypeError(msg)
+
+            out_none = cgutils.is_nonelike(out)
+            if not out_none and not isinstance(out, types.Array):
+                raise errors.NumbaTypeError('output must be an array')
+
+            array_arr = isinstance(array, types.Array)
+            array_ndim = array.ndim if array_arr else 0
+            tup_m1 = (0,) * (array_ndim - 1)
+            indices_arr = isinstance(indices, types.Array)
+            dt = array.dtype if cgutils.is_nonelike(dtype) else dtype
+
+            need_cast = not (array_arr and indices_arr)
+
+            if indices_arr and indices.ndim != 1:
+                msg = ('Expect "indices" array to have at most 1 dimension. '
+                       f'Got {indices.ndim}')
+                raise errors.NumbaValueError(msg)
+
+            if need_cast:
+                def impl_cast(ufunc, array, indices, axis=0, dtype=None, out=None):  # noqa: E501
+                    return ufunc.reduceat(np.asarray(array),
+                                          np.asarray(indices),
+                                          axis,
+                                          dtype=dtype,
+                                          out=out)
+                return impl_cast
+
+            @register_jitable
+            def tuple_slice(tup, pos):
+                s = tup_m1  # TODO: replace this by a literally call
```

Comment:
Does this TODO needs to be addressed ? If not, can we remove it?

Reply:
I'll remove it.